packages/components/src/Filter/FilterBar/subcomponents/AddFiltersMenu/AddFiltersMenu.tsx

```diff
@@ -14,13 +15,22 @@ export const AddFiltersMenu = (): JSX.Element => {
       "Menu button label to show additional available filter options",
   })
 
-  const { getInactiveFilters, showFilter } = useFilterBarContext()
+  const { getInactiveFilters, showFilter, focusId, setFocus } =
+    useFilterBarContext()
   const inactiveFilters = getInactiveFilters()
 
+  useEffect(() => {
+    if (focusId === "add_filter") {
+      buttonRef.current?.focus()
+      setFocus(undefined)
+    }
+  }, [focusId])
```

Comment:
No notes but just to to talk throw my understanding so far, when the add_filter event is dispatched this will refocus to the button and set the focus in in the FilterBarContnent as part of some clean up?

Reply:
yeah exactly, it sets the focus to undefined to reset it back to its original state now that it's done the job we wanted